src/main/kotlin/Main.kt

```diff
@@ -74,27 +78,116 @@ fun main() = application {
             ) {
 
                 val currentColorPalette = MaterialTheme.colorScheme
-
+                var shouldShowFileChooser by remember { mutableStateOf(false) }
                 ComponentScope(
                     onColorPicked = { colorName, color ->
                         val updatedColorPalette = updateColorPalette(
-                                currentColorPalette = currentColorPalette,
-                                colorName = colorName,
-                                color = color,
-                            )
+                            currentColorPalette = currentColorPalette,
+                            colorName = colorName,
+                            color = color,
+                        )
 
                         if (darkmode) {
                             darkColorScheme = updatedColorPalette
                         } else {
                             lightColorScheme = updatedColorPalette
                         }
+                    },
+                    onPrintColors = {
+                        shouldShowFileChooser = true
                     }
                 )
+
+                if (shouldShowFileChooser) {
+                    FileChooserDialog(
+                        title = "Select directory to save file",
+                        onResult = {
+                            saveColorsToFile(
+                                currentColorPalette, file = it, onResult = { success, message ->  
+                                   // Implement success / failure screens for this when VM is implemented,
+                                   // and we can easier manage states.
+                                }
+                            )
+                            shouldShowFileChooser = false
+                        },
+                        onCancel = {
+                            shouldShowFileChooser = false
+                        }
+                    )
+                }
             }
         }
     }
 }
 
+@Composable
+private fun FileChooserDialog(
+    title: String,
+    onResult: (result: File) -> Unit,
+    onCancel: () -> Unit
+) {
+    val fileChooser = JFileChooser(FileSystemView.getFileSystemView())
+    fileChooser.currentDirectory = File(System.getProperty("user.dir"))
+    fileChooser.dialogTitle = title
+    fileChooser.fileSelectionMode = JFileChooser.DIRECTORIES_ONLY
+    fileChooser.isAcceptAllFileFilterUsed = true
+    fileChooser.selectedFile = null
+    fileChooser.currentDirectory = null
+    if (fileChooser.showOpenDialog(null) == JFileChooser.APPROVE_OPTION) {
+        val file = fileChooser.selectedFile
+        onResult(file)
+    } else {
+        onCancel.invoke()
+    }
+}
+
+private fun saveColorsToFile(
+    currentColorPalette: ColorScheme,
+    file: File,
+    onResult: (success: Boolean, message: String) -> Unit
```

Comment:
optional: Maybe convert this to a `sealed class FileResult` which you can reuse for later? Then you dont need to check for Boolean state and then if it has a message